### ISSUE TITLE
Translations failing  when page is idle for couple of hours

### ISSUE THREAD

**Issue by PreethiVS21** [i18next/next-i18next#2267]
## 🐛 Bug Report

in next-i18next version 13.2.2, When translations are rendered using getServerSideProps (next 13.5-pages router)  , translations doesn't happen after couple of hours. I can only see key names . 

## To Reproduce
next-i18next.config.js
```js
const path = require('path');
module.exports = {
  debug: false,
  i18n: {
    defaultLocale: "en",
    locales: ["en", "es"],
  },
  localePath:
    typeof window === 'undefined'
      ? path.resolve('./public/locales')
      : '/locales'
};
```
index.tsx
```
export const getServerSideProps: GetServerSideProps<Props> = async ({
  locale,
})=>{

  return {
    props: {
      ...(await serverSideTranslations(locale ?? 'en', [
        'common',
        'randomns'
      ]))
    }
  };
}
```

## Expected behavior

Translations shouldn't fail intermittently 

## Your Environment

    "i18next": "^22.4.13",
    "next": "^13.5.6",(pages router)
    "next-i18next": "^13.2.2",

**Comment by adrai** [comment]
Hearing this for the first time...
Maybe a problem of your deployment environment?
Or is this reproducible also locally?
Please provide more information.

**Comment by PreethiVS21** [comment]
Yes, I can reproduce it locally. Similar comment was addressed here. Can you confirm if this is needed whenever we need to get translations from getServerSideProps 
https://github.com/vercel/next.js/issues/46730#issuecomment-1489654441

**Comment by adrai** [comment]
seems the way Next.js handles this... yes

**Comment by Sarawut-keng** [comment]
@PreethiVS21 Have you figured out how to fix it? I have also encountered this issue.

**Comment by jgregory-apogee** [comment]
I've found I can reproduce this locally fairly easily.  In our case, we are using a custom Docker image.  Simply rebuilding the image and redeploying it is enough to break the translations.

Steps:
1. Deploy the app in prod mode and open any page.
2. Rebuild / redeploy the image.
3. Soft-navigate from the already-opened page to another page.  Translations will be broken.

I can see that Next is fetching a translations file on every page transition.  For instance, every time I navigate to Page A, it will fetch `/_next/data/tfbRnBfT3Y6OXf7Afbrm1/en/PageA/6632778e658b8005459f8503.json`, and every time I go to Page B, it will fetch `/_next/data/tfbRnBfT3Y6OXf7Afbrm1/en/PageB/99125246-21fc-4742-876c-e7d3b9a52bcc.json`.  But after a redeploy, those two URLs return an empty translation list: `{}`.  Presumably this is because the Next `BUILD_ID` has changed.  If I do a full reload, the pages fetch slightly different URLs, like: `/_next/data/eCtPEywIx-3IoSgG-JHVF/en/PageA/6632778e658b8005459f8503.json`.  (Note that the old `BUILD_ID` `tfbRnBfT3Y6OXf7Afbrm1` has been replaced with `eCtPEywIx-3IoSgG-JHVF`).

Anyways, it seems I need a way to detect that there's a new `BUILD_ID` in play and force next to do a full-page-load transition.

**Comment by BrilliantDecision** [comment]
@PreethiVS21 @Sarawut-keng any updates guys? same issue on the same next version 13.5.6 (pages router)

**Comment by BrilliantDecision** [comment]
@adrai getServerSideProps totally doesnt work after couple of hours, its ok with getStaticProps

**Comment by adrai** [comment]
> @adrai getServerSideProps totally doesnt work after couple of hours, its ok with getStaticProps

If that's the case you may ask the Next.js team...

**Comment by Sarawut-keng** [comment]
@BrilliantDecision after reviewing the case, I have figured out that @jgregory-apogee  ’s comment was correct. After my CD re-deployment, the language broke. I am not sure if this occurs only on the page router. Right now, we do not have the best solution yet. We have just briefed users that we have updated to a new version and that they need to refresh the screen.

**Comment by BrilliantDecision** [comment]
@Sarawut-keng and how do you implement this notification, what do you use for this? how do you understand that you need to refresh the page? some error catcher?

**Comment by mehmetyilmaz001** [comment]
We have the exact same problem here. i18n store is empty on after some deployments even the translation files are exist.

**Comment by allicanseenow** [comment]
Does this mean this library is not ready for production, as this is a pretty common use case potentially for all the next.js apps?

**Comment by BrilliantDecision** [comment]
@allicanseenow library is good, but next js has an issue, and it was fixed in next js 14.. if I’m not mistaken
